lib/src/model/annotation.dart

```diff
@@ -36,14 +38,46 @@ final class Annotation extends Attribute {
     var parameterText =
         source.substring(startIndex == -1 ? source.length : startIndex);
 
-    return '@$linkedName${const HtmlEscape().convert(parameterText)}';
+    var escapedParameterText = const HtmlEscape().convert(parameterText);
+    return '@$linkedName$_linkedTypeArguments$escapedParameterText';
   }
 
   @override
-  String get linkedName => _annotation.element is PropertyAccessorElement
-      ? _packageGraph.getModelForElement(_annotation.element!).linkedName
-      // TODO(jcollins-g): consider linking to constructor instead of type?
-      : _modelType.linkedName;
+  String get linkedName => switch (_annotation.element) {
+        PropertyAccessorElement element =>
+          _packageGraph.getModelForElement(element).linkedName,
+        ConstructorElement element =>
+          _packageGraph.getModelForElement(element).linkedName,
+        _ => _modelType.linkedName
+      };
+
+  /// The linked type argument text, with `<` and `>`, if there are any type
+  /// arguments.
+  String get _linkedTypeArguments {
+    var element = _annotation.element;
+    if (element is! SubstitutedConstructorElementImpl) {
```

Comment:
Hi @scheglov , I had to use `SubstitutedConstructorElementImpl` here, in order to get the type arguments on the constructor call of an annotation.

I don't see another way to get the information from the ElementAnnotation. This does feel hacky though, walking through the `substitution` field on private API.